Assets/Scripts/StandardSamples/UI/Common/SampleSelectableStateHandler.cs

```diff
@@ -0,0 +1,108 @@
+* 
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+* 
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE.
+*/
+
+namespace PlayEveryWare.EpicOnlineServices.Samples
+{
+    using System.Collections;
+    using UnityEngine;
+    using UnityEngine.UI;
+    using UnityEngine.EventSystems;
+    using System;
+
+    public class SampleSelectableStateHandler : MonoBehaviour
+    {
+        public struct SampleInteractableNewState
+        {
+            public bool Interactable;
+            public string NewTooltipText;
+
+            public SampleInteractableNewState(bool shouldBeInteractable, string newTooltipText = "")
+            {
+                Interactable = shouldBeInteractable;
+                NewTooltipText = newTooltipText;
+            }
+        }
+
+        public delegate SampleInteractableNewState GetSampleInteractable();
+
+        private static event Action UpdateSelectableStates;
+
+        private GetSampleInteractable sampleInteractableFunction { get; set; }
+
+        [SerializeReference]
+        private UITooltip attachedTooltip;
+
+        [SerializeReference]
+        private Selectable attachedSelectable;
+
+        public void SetSampleInteractableAction(GetSampleInteractable inSampleInteractableFunction)
+        {
+            sampleInteractableFunction = inSampleInteractableFunction;
+            UpdateFromFunction();
+        }
+
+        private void OnEnable()
+        {
+            UpdateSelectableStates += UpdateFromFunction;
+        }
+
+        private void OnDisable()
+        {
+            UpdateSelectableStates -= UpdateFromFunction;
+        }
+
+        void UpdateFromFunction()
+        {
+            if (sampleInteractableFunction == null)
+            {
+                return;
+            }
+
+            SampleInteractableNewState newState = sampleInteractableFunction();
+
+            if (!string.IsNullOrEmpty(newState.NewTooltipText) && attachedTooltip != null)
+            {
+                if (newState.Interactable)
+                {
+                    // TODO: Restore the original tooltip text if it is interactable
+                    // At the moment, none of the buttons that this was applied to
+                    // had any tooltips to begin with
+                    attachedTooltip.Text = string.Empty;
+                }
+                else
+                {
+                    attachedTooltip.Text = newState.NewTooltipText;
+                }
+            }
```

Comment:
Am I correct to infer from this that what this PR does is remove the tooltip from buttons that should not be selectable?

Reply:
The opposite! This PR:

- Removes tooltips from buttons that are currently Interactable
- Adds tooltips back to things that have become non-interactive
- Determine what tooltip text to use in the above case
- Uses the attached functions to determine if it should be active or not